### DISCUSSION TITLE
add `onIntentSubmit` option

### DISCUSSION THREAD

**Discussion by AkifumiSato** [edmundhung/conform#563] [Ideas]
I am working on a library [location-state](https://github.com/recruit-tech/location-state). It is a state management library that synchronizes with the history location.
The `useForm` has an `onSubmit` option, but there is no way to hook Intent Button-derived submissions. My implementation would be much easier with `onIntentSubmit`.

```ts
// `meta.name` is intent name.
type OnIntentSubmit = (React.FormEvent<HTMLFormElement>, meta: { name: string }) => void

// use case example
const [form, fields] = useForm({
  onIntentSubmit(event, { meta: { name } }) {
    if (name === "insert") {
      ...
    }
  },
  ...
});
```

**Comment by edmundhung** [comment]
Hi @AkifumiSato, I would like to understand more how you will use this information for. Can you share some more details?

**Reply by AkifumiSato** [reply]
@edmundhung Thank you for your reply! I share the code currently being implemented.
https://github.com/recruit-tech/location-state/blob/fd754dc61fb776042add533e9d7e55b769ad63b1/apps/example-next-conform/src/lib/use-location-form.ts#L142-L151

Need to get and save data when there is a change in the form.
Currently wraps onSubmit from getFormProps.
I can't determine the type of intent or data, so I re-render it and use useEffect to take care of these things.(https://github.com/recruit-tech/location-state/blob/fd754dc61fb776042add533e9d7e55b769ad63b1/apps/example-next-conform/src/lib/use-location-form.ts#L113-L116)
With the proposed API, these implementations can be made very simple.
